cve_bin_tool/cvedb.py

```diff
@@ -763,18 +765,23 @@ def update_vendors(self, cve_data):
 
         return updated_severity, updated_affected
 
-    def db_open_and_get_cursor(self) -> sqlite3.Cursor:
-        """Opens connection to sqlite database, returns cursor object."""
-
-        if not self.connection:
-            self.connection = sqlite3.connect(self.dbpath)
-        if self.connection is not None:
-            cursor = self.connection.cursor()
-        if cursor is None:
-            # if this happens somsething has gone horribly wrong
-            LOGGER.error("Database cursor does not exist")
-            raise CVEDBError
-        return cursor
+    def db_open_and_get_cursor(self):
+    """Open the database connection with retry logic to handle 'database is locked' errors."""
+    retries = 0
+    while retries < MAX_RETRIES:
+        try:
+            self.connection = sqlite3.connect(self.dbpath, timeout=10)  # Extend timeout
+            self.connection.execute("PRAGMA journal_mode=WAL;")  # Improve concurrency
```

Comment:
Trying to think of side effects this might cause. This will create some extra files on disk next to our database, right? I don't think there would be issues with this change, but I defer to the other maintainers who might be more familiar with the SQLite stuff...